src/Guzzle6HttpAdapter.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace Http\Adapter;
+
+use FOS\HttpCache\Exception\ProxyUnreachableException;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Exception\TransferException;
+use GuzzleHttp\Pool;
+use GuzzleHttp\Promise\Promise;
+use Http\Adapter\Common\Exception\HttpAdapterException;
+use Http\Adapter\Common\Exception\MultiHttpAdapterException;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class Guzzle6HttpAdapter implements PsrHttpAdapter
+{
+    public function __construct(Client $client = null)
+    {
+        $this->client = $client ?: new Client();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function sendRequest(RequestInterface $request)
+    {
+        try {
+            return $this->client->send($request);
+        } catch (TransferException $e) {
+            throw ProxyUnreachableException::proxyUnreachable($request, $e);
```

Comment:
I am not sure I understand why you throw this exception here. `sendRequest` should throw an `HttpAdapterException`.